### DISCUSSION TITLE
How to find peptides from a dataset of 30.000 molecules?

### DISCUSSION THREAD

**Discussion by potatotatalicious** [rdkit/rdkit#6770] [Q&A]
I have built a dataset of 30.000 molecules (smiles) using several databases. Now I want to identify the peptides, how can I do it? Thank you!

**Comment by kienerj** [comment]
substructure search for peptide bond? And then see how many results you got and if you need to apply additional filter or the list is small enough for manual check.

What have you tried so far that isn't working?

**Comment by dehaenw** [comment]
You may apply an initial SMARTS filter to check for the presence of at least one amino acid backbone.
```python
from rdkit import Chem
# some array of smiles, make sure these are parsable smiles
smis = ["c1ccccc1","NC(C)C(=O)NCC(=O)O","ClC(Cl)(Cl)"]
patt=Chem.MolFromSmarts("[NX3;H1,H2]-[CX4;H1,H2]-[CX3]=[OX1]")
peptide_smis=[s for s in smis if Chem.MolFromSmiles(s).HasSubstructMatch(patt)==True]
# filtered array with only smiles 
print(peptide_smis)
```

this outputs `['NC(C)C(=O)NCC(=O)O']`. After this filter depending on the number of molecules it is probably to manually annotate or to confirm they are actually peptides in a more sophisticated way
